c7n/resources/emr.py

```diff
@@ -332,6 +333,52 @@ def augment(self, resources):
         return resources
 
 
+@EMRSecurityConfiguration.filter_registry.register('kms-key')
+class EmrSecurityConfigurationKmsFilter(KmsRelatedFilter):
+    """
+    Filter a resource by its associcated kms key and optionally the alias name
+    of the kms key by using 'c7n:AliasName'
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: emr-security-configuration-kms-key
+                resource: aws.emr-security-configuration
+                filters:
+                  - type: kms-key
+                    key: c7n:AliasName
+                    value: "^(alias/aws/)"
+                    op: regex
+    """
+
+    RelatedIdsExpression = 'SecurityConfiguration.EncryptionConfiguration.AtRestEncryptionConfiguration.\
+                            LocalDiskEncryptionConfiguration.AwsKmsKey'
+
+    related_ids = []
+
+    def get_related_ids(self, resources):
+        if self.related_ids:
+            return self.related_ids
+
+        alias_names = set(jmespath.search(
+            "[].%s" % self.RelatedIdsExpression, resources))
+        resource_manager = self.get_resource_manager()
```

Comment:
why is get-related-ids being overriden here, its quite strange to see it do a fetch on the resources, that's normally done in get-related which looks like it was inlined here. get-related-ids gets called w/ different arguments then the full set of resources, but the way this is caching looks like it would be incorrect with different parameters. so first question is why the override and inline fetch, is that to resolve alias to ids? most of the kms api will work with either alias or id.

Reply:
It's to resolve alias arns to ids. From looking at the KmsRelatedFilter, https://github.com/cloud-custodian/cloud-custodian/blob/7d2c2b688564e971ba9faf0dce4144a3d2162caf/c7n/filters/kms.py, it seems like I need to transform these alias arns to ids, and I wasn't sure how to do that without the override and inline fetch.

Reply:
you can omit this override, the actual aws api supports many parameter styles (arns, ids, aliases, etc) see DescribeKey docs for a full listing.